.chloggen/mc_spanmetrics.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: spanmetricsconnector
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add a feature gate to use the latest semantic conventions for the status code attribute on generated metrics.
```

Comment:
nit: We can use `subtext` to explain the rationale behind introducing this breaking change, as well as how to use feature gates to roll it back.

In `note` we explicitly say that the key of span status change from `status.code` to `otel.status_code`